src/environment.js

```diff
@@ -324,7 +325,9 @@ var Environment = Obj.extend({
 
         var tmpl = new Template(src, this, opts.path);
         return tmpl.render(ctx, cb);
-    }
+    },
+
+    waterfall: waterfall
```

Comment:
Eh, in JS the distinction is pretty murky anyway (just a matter of `this` binding). Doesn't bother me.